chapters/functions.tex

```diff
@@ -295,8 +296,8 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
 A function having a formal function parameter that is \lstinline!impure! must be marked \lstinline!pure! or \lstinline!impure!.
 
 \begin{nonnormative}
-Comment: The semantics are undefined if the function call of an
-impure function is part of an algebraic loop.
+Comment: The semantics are undefined if the function call of an impure function is part of an algebraic loop.
+This can occur for the deprecated case of external functions assumed to be impure, or when using \lstinline!pure(expression)! to call an impure function in a pure function.
```

Comment:
```suggestion
Comment: The exact semantics are undefined if the function call of an impure function is part of an algebraic loop.
Specifically the number of calls with external side-effects is unspecified.
However, for impure functions where the outputs only depend on the inputs the algebraic loop should be solved correctly.
This can occur for the deprecated case of external functions assumed to be impure, or when using \lstinline!pure(expression)! to call an impure function in a pure function.
```
I believe this was the intended idea.

Not a general undefined-behavior in C-terms (with nasal demons etc), but just the pragmatic point that you cannot solve an algebraic loop with an impure function that e.g., returns the current time in nano-seconds, and that you have no control of the number of external side-effects generated.

That's why it is non-normative.

Reply:
How about speaking of _side-effect semantics_?  Wouldn't that then be nicely elaborated by the outputs-only-depending-on-inputs rule?

Reply:
Ok, makes sense.

Reply:
Have now done that.